sources/packages/backend/apps/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -117,9 +118,20 @@ export class StudentAppealStudentsController extends BaseController {
     let dryRunSubmissionResults: DryRunSubmissionResult[] = [];
     try {
       const dryRunPromise: Promise<DryRunSubmissionResult>[] =
-        payload.studentAppealRequests.map((appeal) =>
-          this.formService.dryRunSubmission(appeal.formName, appeal.formData),
-        );
+        payload.studentAppealRequests.map((appeal) => {
+          // Check if the form has any non submission data(data which is not persisted but required for validation)
+          // which needs to be populated at the server side for dry run submission.
+          if (appeal.formData instanceof StudentAppealNonSubmissionData) {
+            appeal.formData = {
+              ...appeal.formData,
+              programYear: application.programYear.programYear,
+            };
+          }
```

Comment:
Correct me if I'm wrong. Since you checked off `protected` for `programYear`, it will be returned when queried via API. So `programYear` is part of the payload, and it's not necessary to get the data again from the database here.

Reply:
Using the program `programYear` from the payload would open the API to the user to manipulate the information.
Data submitted to the API should be only data provided by the user. Any other value should have the server data as a source of truth. Please me know if further clarification is needed.